src/resources/views/base/inc/alerts.blade.php

```diff
@@ -1,21 +1,48 @@
 {{-- Bootstrap Notifications using Prologue Alerts & PNotify JS --}}
 <script type="text/javascript">
-  Noty.overrideDefaults({
-    layout   : 'topRight',
-    theme    : 'backstrap',
-    timeout  : 2500, 
-    closeWith: ['click', 'button'],
-  });
 
-  @foreach (\Alert::getMessages() as $type => $messages)
+    // This is intentionaly run after dom loads so this way we can avoid showing duplicate alerts
+    // when the user is beeing redirected by persistent table, that happens before this event triggers.
 
-      @foreach ($messages as $message)
+    window.addEventListener('DOMContentLoaded', function() {
 
-        new Noty({
-          type: "{{ $type }}",
-          text: "{!! str_replace('"', "'", $message) !!}"
-        }).show();
 
-      @endforeach
-  @endforeach
-</script>
\ No newline at end of file
+        Noty.overrideDefaults({
+                layout   : 'topRight',
+                theme    : 'backstrap',
+                timeout  : 2500,
+                closeWith: ['click', 'button'],
+            });
+
+        // get alerts from the alert bag
+        var $alerts_from_php = JSON.parse('@json(\Alert::getMessages())');
+
+        // get the alerts from the localstorage
+        var $alerts_from_localstorage = localStorage.getItem('backpack_alerts');
+
+        if($alerts_from_localstorage !== null) {
+            $alerts_from_localstorage = JSON.parse($alerts_from_localstorage);
+        }
```

Comment:
```suggestion
        var $alerts_from_localstorage = JSON.parse(localStorage.getItem('backpack_alerts'));
```

I think it's safe to do this directly, `JSON.parse` deals good with nulls.

Reply:
Also, should it be set by default to empty object? `?? {}`

Reply:
Totally, just fixed!